lua/telescope/_extensions/ghcli.lua

```diff
@@ -1,192 +1,12 @@
-  opts = opts or {}
-  opts.limit = opts.limit or 100
-  local opts_query = parse_opts(opts , 'issue')
-  local cmd = 'gh issue list '.. opts_query
-  local results = vim.split(utils.get_os_command_output(cmd), '\n')
-  pickers.new(opts, {
-    prompt_title = 'Issues',
-    finder = finders.new_table {
-      results = results,
-      entry_maker = make_entry.gen_from_string(opts),
-    },
-    previewer = previewers.new_termopen_previewer{
-      get_command = function(entry)
-        local tmp_table = vim.split(entry.value,"\t");
-        if vim.tbl_isempty(tmp_table) then
-          return {"echo", ""}
-        end
-        return { 'gh' ,'issue' ,'view',tmp_table[1] }
-      end
-    },
-    sorter = conf.file_sorter(opts),
-    attach_mappings = function(_,map)
-      actions.goto_file_selection_edit:replace(actions.close)
-      map('i','<c-t>',GH.a_gh_web_view('issue'))
-      return true
-    end
-  }):find()
-end
-
-GH.b_gh_pull_request = function(opts)
-  opts = opts or {}
-  opts.limit = opts.limit or 100
-  local opts_query = parse_opts(opts , 'pr')
-  local cmd = 'gh pr list ' .. opts_query
-  local results = vim.split(utils.get_os_command_output(cmd) , '\n')
-  pickers.new(opts, {
-    prompt_title = 'Pull Requests' ,
-    finder = finders.new_table {
-      results = results,
-      entry_maker = make_entry.gen_from_string(opts),
-    },
-    previewer = previewers.new_termopen_previewer{
-      get_command = function(entry)
-        local tmp_table = vim.split(entry.value,"\t");
-        if vim.tbl_isempty(tmp_table) then
-          return {"echo", ""}
-        end
-        return { 'gh' ,'pr' ,'view',tmp_table[1] }
-      end
-    },
-    sorter = conf.file_sorter(opts),
-    attach_mappings = function(_,map)
-      actions.goto_file_selection_edit:replace(GH.a_gh_pr_checkout)
-      map('i','<c-t>',GH.a_gh_web_view('pr'))
-      return true
-    end
-  }):find()
-end
 
-GH.b_gh_gist = function(opts)
-  opts = opts or {}
-  opts.limit = opts.limit or 100
-  local opts_query = parse_opts(opts , 'gist')
-  local cmd = 'gh gist list '..opts_query
-  local results = vim.split(utils.get_os_command_output(cmd), '\n')
-  pickers.new(opts, {
-    prompt_title = 'gist list' ,
-    finder = finders.new_table {
-      results = results,
-      entry_maker = make_entry.gen_from_string(opts),
-    },
-    previewer = GH.p_gh_gist_preview.new(opts),
-    sorter = conf.file_sorter(opts),
-    attach_mappings = function(_,map)
-      actions.goto_file_selection_edit:replace(GH.a_gh_gist_append)
-      map('i','<c-t>',GH.a_gh_web_view('gist'))
-      return true
-    end
-  }):find()
-end
 
 return require('telescope').register_extension {
     setup = function()
-      builtin.gh_gist = GH.b_gh_gist
-      builtin.gh_issues = GH.b_gh_issues
-      builtin.gh_pull_request = GH.b_gh_pull_request
+      builtin.gh_gist =gh_b.gh_gist
+      builtin.gh_issues = gh_b.gh_issues
+      builtin.gh_pull_request =gh_b.gh_pull_request
```

Comment:
Do not override `builtin` from a plugin.

You should expose them in `exports` and plugins can call your functions from there.

Reply:
Hi i override the builtin because we can autocomplete it on command mode with `:Telescope gh_gist` .
and we can use options on command mode `Telescope gh_issues author=windwp label=bug`

Reply:
@windwp I am about to merge the correct fix for that into telescope core. https://github.com/nvim-telescope/telescope.nvim/pull/380
After that is merged can you `export` your functions and no longer inject it into builtin thanks :)